frontend/src/components/create-user-page/CreateUser.tsx

```diff
@@ -61,49 +98,81 @@ export const CreateUser: React.FunctionComponent = () => {
             `${loginContext.username}:${loginContext.password}`
           )}`,
         },
-        body: JSON.stringify(FormData),
-      }).then((res) => res.json());
+        body: JSON.stringify({
+          name: formName,
+          username: formUsername,
+          email: formEmail,
+          password: formPassword,
+        }),
+      })
+        .then((res) => {
+          setAddUserMessage(
+            res.status == 200
+              ? "User added successfully!"
+              : "Sorry, Couldn't add the user!"
+          );
+        })
+        .then(() => (e.target as HTMLFormElement).reset());
     }
   };
 
   return (
-    <div>
+    <div className="form">
       <form onSubmit={handleSubmit}>
-        <label>
-          Name:
-          <input
-            type="text"
-            name="Name"
-            onChange={(e) => setFormName(e.target.value)}
-          />
-        </label>
-        <label>
-          Username:
-          <input
-            type="text"
-            name="Username"
-            onChange={(e) => setFormUsername(e.target.value)}
-          />
-        </label>
-        <label>
-          Email:
-          <input
-            type="email"
-            name="Email"
-            onChange={(e) => setFormEmail(e.target.value)}
-          />
-        </label>
-        <label>
-          Password:
-          <input
-            type="password"
-            name="Password"
-            onChange={(e) => setFormPassword(e.target.value)}
-            onBlur={(e) => validatePassword(e.target.value)}
-          />
-        </label>
-        <input type="submit" value="Create User" />
+        <h1 className="form_heading">Create New User</h1>
+        <div className="form_field">
+          <div className="form_field--row">
+            <label className="form_field--label">Name:</label>
+            <input
+              className="form_field--input"
+              type="text"
+              name="Name"
+              onChange={(e) => validateName(e.target.value)}
+            />
+          </div>
+          <p className="message">{nameMessage}</p>
+        </div>
```

Comment:
This is a bit too many `<div>`s for my comfort - could you instead (for example) style the `<form>` with using flex _column_, then have the `<input>` _inside_ the `<label>` (as is [sometimes standard](https://stackoverflow.com/questions/774054/should-i-put-input-elements-inside-a-label-element)), meaning you can have identical styling but with no `<div>`s at all!

You may want to use things like [`:first-child`](https://www.w3schools.com/cssref/sel_firstchild.php) to style that initial text node, alternatively you can do as this guy says:
![image](https://user-images.githubusercontent.com/4410524/197532578-4b933728-1ba7-4c02-b64e-def894f58583.png)

Reply:
OK, thanks, I have gone back and edited it quite a lot based on this, but hopefully it is more suitable!